src/java.desktop/share/classes/javax/swing/text/html/FormView.java

```diff
@@ -273,13 +273,14 @@ private JComponent createInputComponent(AttributeSet attr, Object model) {
             maxIsPreferred = 3;
         } else if (type.equals("image")) {
             String srcAtt = (String) attr.getAttribute(HTML.Attribute.SRC);
+            String altAtt = (String) attr.getAttribute(HTML.Attribute.ALT);
             JButton button;
             try {
                 URL base = ((HTMLDocument)getElement().getDocument()).getBase();
                 @SuppressWarnings("deprecation")
                 URL srcURL = new URL(base, srcAtt);
-                Icon icon = new ImageIcon(srcURL);
-                button  = new JButton(icon);
+                ImageIcon icon = new ImageIcon(srcURL);
+                button = icon.getImageLoadStatus() == MediaTracker.COMPLETE ? new JButton(icon) : new JButton(altAtt);
             } catch (MalformedURLException e) {
                 button = new JButton(srcAtt);
```

Comment:
```suggestion
                button = new JButton(altAtt);
```
If `altAtt` is provided, it should be used to handle invalid `srcAtt` too.

Reply:
@aivanov-jdk My apologies, I forgot to reply. I wanted to distinguish the two cases : (Missing resource or bad resource name) and malformed URL. Like in the previous example I gave.

Reply:
> @aivanov-jdk My apologies, I forgot to reply. I wanted to distinguish the two cases : (Missing resource or bad resource name) and malformed URL. Like in the previous example I gave.

What purpose does this distinction serve?

It could give an additional insight to the app developer. At the same time, the app developer should ensure the image loads.

Does the user of the app care about it \[the distinction\]? Unlikely. The end result is the image isn't loaded. As HTML specifies, the `alt` attribute contains the image description, which is more useful to the user in _all_ the cases where the image failed to load.

Therefore I think you should fallback to `altAtt` if it's provided.

This actually raises another question: what if `altAtt` isn't provided? Should it fallback to `srcURL`?

Reply:
> What purpose does this distinction serve?

Definitely an application developer and because it's the context of this HTML implementation, may I preserve this distinction.

> This actually raises another question: what if altAtt isn't provided? Should it fallback to srcURL?

Yes, it's a logical error on my part.

Reply:
![image](https://github.com/openjdk/jdk/assets/19194678/0c4ba14d-84df-4c45-abb2-f3f9dca01bd2)


```
            <html>
                <body>
                    <input type=image name=point src="file:oracle_logo_50x50.jpg" alt="Logo Oracle JPG">
                    <p>
                    <input type=image name=point src="file:oracle_logo_50x50.jpg">
                    <p>
                    <input type=image name=point src="file:none_oracle_logo_50x50.jpg" alt="Logo Oracle JPG">
                    <p>
                    <input type=image name=point src="file:none_oracle_logo_50x50.jpg">
                    <p>
                    <input type=image name=point src="files:none_oracle_logo_50x50.jpg" alt="Logo Oracle JPG">
                    <p>
                    <input type=image name=point src="files:none_oracle_logo_50x50.jpg">
                    <p>
                </body>
            </html>
```

- Left before the Patch
- Right with your suggestion and if the distinction is preserved.

Reply:
Yes, that's what I meant.

A button with the plain source is better than a button without a title at all.

Reply:
> Are you going to address it?

Yes.